### ISSUE TITLE
False positive: Annotated type-hints passed to Type

### ISSUE THREAD

**Issue by BrianPugh** [microsoft/pyright#7817]
**Describe the bug**
At some point within the past few weeks, passing in an `Annotated` type to a parameter annotated as `Type` generates an error.

```python
from typing import Annotated, Type


def foo(type_: Type):
    pass


foo(int)  # This is fine
foo(Annotated[int, "bar"])  # error: Argument of type "type[int]" cannot
                            # be assigned to parameter "type_" of type
                            # "Type[Unknown]" in function "foo"
```

**VS Code extension or command-line**
command-line v1.1.360

**Comment by erictraut** [comment]
This isn't a false positive. The new behavior is correct.

This behavior was recently clarified in the [Python typing spec](https://typing.readthedocs.io/en/latest/spec/qualifiers.html#id4), and pyright was updated to conform to the spec. 

Here's the exact wording from the spec:
> As with most [special forms](https://typing.readthedocs.io/en/latest/spec/glossary.html#term-special-form), Annotated is not type compatible with type or type[T]:

If you're interested in the discussion that led to this resolution, here's a [link](https://discuss.python.org/t/is-annotated-compatible-with-type-t/43898).

**Comment by BrianPugh** [comment]
Thanks!
